test/results_test.exs

```diff
@@ -0,0 +1,94 @@
+defmodule Wanda.ResultsTest do
+  use ExUnit.Case
+  use Wanda.DataCase
+
+  import Wanda.Factory
+
+  alias Wanda.Catalog
+  alias Wanda.Results
+
+  alias Wanda.Execution.{
+    Evaluation,
+    ExecutionResult,
+    Fact
+  }
+
+  describe "Appending Execution Results to the history log" do
+    test "should correctly add an item to an empty log" do
+      gathered_facts = %{
+        "expect_check" => %{
+          "agent_1" => [
+            %Fact{name: "corosync_token_timeout", check_id: "expect_check", value: 30_000}
+          ]
+        }
+      }
+
+      execution_id = UUID.uuid4()
+      group_id = UUID.uuid4()
+      checks = [Catalog.get_check("expect_check")]
+
+      result = Evaluation.execute(execution_id, group_id, checks, gathered_facts)
```

Comment:
Should we depend on external functions to test? I think we should create the result object directly, to avoid false negatives if the `execute` (which involves a lot of stuff) changes

Reply:
i agree with @arbulu89 here we should not depend on any external function as this is not a wide test.
we should instead craft a result struct with a factory or similar and see if the stored result is what we expect

Reply:
Right, I cut it too short :sweat_smile: my bad.

Fixed in the last commit.